### DISCUSSION TITLE
Filter nodes in Remote Scene Tree updates

### DISCUSSION THREAD

**Discussion by firescaleVR** [godotengine/godot-proposals#10900] [Editor]
The remote scene tree viewer becomes unresponsive when there are too many nodes in a scene. The scene runs perfectly fine, 200-300 fps, but the viewer and editor become almost frozen because they are updating nodes like rocks and plants that do not need to be inspected. 

Why is this an issue? Because there are many scatter addons that fill the scene tree with multimesh containers and the nodes they instance and people rely on these addons to add detail to their scenes. 

There needs to be some way for nodes or groups to opt-in to the remote scene tree viewer so that it only reports on things that developers are actively trying to debug.

**Comment by theraot** [comment]
There is a filter (by name, by type, by group), is it not working or not sufficient?

![image](https://github.com/user-attachments/assets/c354b35c-7036-4520-8522-f713ef06200a)

**Reply by firescaleVR** [reply]
That only hides the other nodes in the tree, it's still checking the hidden nodes for updates. It does nothing to improve the performance of the remote viewer.
